### PR TITLE
Added Algorithms

### PR DESCRIPTION
**Pull request by KrishnaAlagiri**
## Changes Made:
- Added algorithm for December 20 
- Added algorithm for December 21

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -442,23 +444,78 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+    ```
+  - **Resources**  
+    [Dijkstra's Algorithm](https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm)
+
+
+## **December 21 - Currency Convertor**
+  - **Problem**
+      -  Calculate the currency exchange rates between two user input countries using the attached **.csv** file in the reference.
+      - ``Note:`` All the given ``VALUES`` in the csv file are a countries equivalent value in USD.
+      - ``Optional:`` Generate a complete list of currency units for conversion
```
Comment (akashsara) [inline README.md]: What do you mean by this?

File: README.md [hunk 2]
```diff
@@ -442,23 +444,78 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
-    
+
   - **Resources**
     - [Hash function – Wikipedia](https://en.wikipedia.org/wiki/Hash_function)
     - [Hash table - Wikipedia](https://en.wikipedia.org/wiki/Hash_table)
     - [An in-depth video about cryptocurrencies, blockchain and hash functions](https://youtu.be/bBC-nXj3Ng4)
-    
+
+
+### **December 20 - Dijkstra's Shortest Reach**
```
Comment (akashsara) [inline README.md]: Shortest Path* no?

File: README.md [hunk 3]
```diff
@@ -442,23 +444,78 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+    1                4
+    2                12
+    3                19
+    4                21
+    5                11
+    6                9
+    7                8
+    8                14
+    ```
+  - **Resources**
```
Comment (akashsara) [inline README.md]: We're gonna need way more resources :P

I think this is the hardest program in the list so far. Need an explanation of what a graph is, what the notation in the example means, maybe a visual explanation of how Dijkstra's Algorithm works [https://visualgo.net/en/sssp or https://www.cs.usfca.edu/~galles/visualization/Dijkstra.html]

File: README.md [hunk 4]
```diff
@@ -354,7 +356,7 @@ We have a small collection of algorithms, one for every day of the month. Scroll
     - [More about pascals triangle and how they work](https://www.youtube.com/watch?v=YUqHdxxdbyM)
 
 
-### **December 16 - Find Temperature Difference Between Two Cities**
+### **December 16 - Find Temperature Difference Between Two     Cities**
```
[the remark this answers is not in the document]
Comment (KrishnaAlagiri) [reply README.md]: I've got no idea how that <tab> came 😆

File: README.md (hunk 2 above)
Comment (KrishnaAlagiri) [reply README.md]: Both **Dijkstra's Shortest Reach** and **Dijkstra's Shortest Path** mean the same.

File: README.md (hunk 1 above)
Comment (KrishnaAlagiri) [reply README.md]: I forgot to complete the sentence 🙍‍♂️

File: README.md (hunk 3 above)
Comment (KrishnaAlagiri) [reply README.md]: Cool Resources! I'll update them along with a detailed explanation of terminologies involved.

**Comment by KrishnaAlagiri** [comment]
## All Issues Resolved.

File: README.md [hunk 5]
```diff
@@ -445,24 +445,33 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
     - [Hash table - Wikipedia](https://en.wikipedia.org/wiki/Hash_table)
     - [An in-depth video about cryptocurrencies, blockchain and hash functions](https://youtu.be/bBC-nXj3Ng4)
 
 
 ### **December 20 - Dijkstra's Shortest Reach**
+  - **Terminologies Involved**
+    - A **graph** is a pictorial representation of a set of objects where some pairs of objects are connected by links.
+    - Each node of the graph is represented as a **vertex**.
+    - An **edge** represents a path between two vertices or a line between two vertices.
+    - A **path** represents a sequence of edges between the two vertices.
```
Comment (akashsara) [inline README.md]: So these particular definitions are recursive/confusing xD

An edge represents a path. A path represents a sequences of edges. So an edge represents a sequence of edges?

File: README.md [hunk 6]
```diff
@@ -496,14 +505,16 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+    - [Visualisation of Dijkstra's Algorithm #2](https://www.cs.usfca.edu/~galles/visualization/Dijkstra.html)
 
 
 ## **December 21 - Currency Convertor**
   - **Problem**
       -  Calculate the currency exchange rates between two user input countries using the attached **.csv** file in the reference.
-      - ``Note:`` All the given ``VALUES`` in the csv file are a countries equivalent value in USD.
-      - ``Optional:`` Generate a complete list of currency units for conversion
+      - ``Note:`` All the given ``VALUES`` in the csv file are a country's equivalent value in USD.
+      - ``Optional:`` Also, generate a complete list of currency exchange rates between the input currency and the world currencies.
```
Comment (akashsara) [inline README.md]: So for instance if I give the input as India, the program should give a list of exchange rates for India? Like USA = 70, Malaysia = 17 and so on?

Comment (KrishnaAlagiri) [reply README.md]: Yess!

File: README.md (hunk 5 above)
Comment (KrishnaAlagiri) [reply README.md]: Those are my AU textbook definitions xD
I'll change em!

File: README.md (hunk 6 above)
Comment (akashsara) [reply README.md]: Alright, just put that in as an explanation for it. It's a bit vague as it is now.

**Comment by KrishnaAlagiri** [comment]
## All Issues Resolved (I guess)

**Comment by KrishnaAlagiri** [comment]
## All Issues Resolved (I guess)

File: README.md [hunk 7]
```diff
@@ -442,23 +444,89 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+    - [What is Dijkstra's Algorithm?](https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm)
+    - [Visualisation of Dijkstra's Algorithm #1](https://visualgo.net/en/sssp)
+    - [Visualisation of Dijkstra's Algorithm #2](https://www.cs.usfca.edu/~galles/visualization/Dijkstra.html)
+
+
+## **December 21 - Currency Convertor**
+  - **Problem**
+      -  Calculate the currency exchange rates between two user input countries using the attached **.csv** file in the reference.
+      - ``Note:`` All the given ``VALUES`` in the csv file are a country's equivalent value in USD.
+      - ``Optional:`` Also, generate a complete list of currency exchange rates between the input currency and the world currencies.
```
Comment (akashsara) [inline README.md]: Add this to the optional part. "For instance, if I input India, then my output would be a list of each country's exchange rate for India. So USA = 71.93, Malaysia = 17 etc.

File: README.md [hunk 8]
```diff
@@ -442,23 +444,89 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+
   - **Resources**
     - [Hash function – Wikipedia](https://en.wikipedia.org/wiki/Hash_function)
     - [Hash table - Wikipedia](https://en.wikipedia.org/wiki/Hash_table)
     - [An in-depth video about cryptocurrencies, blockchain and hash functions](https://youtu.be/bBC-nXj3Ng4)
-    
+
+
+### **December 20 - Dijkstra's Shortest Reach**
+  - **Terminologies Involved**
```
Comment (akashsara) [inline README.md]: An edge is a connection or line between two vertices.

A path is a sequence of vertices connected together by edges. Every vertex in a path is adjacent to the vertices next to it. 

Two vertices are adjacent if they share a common edge.

Distance is the the distance traveled to reach one vertex from another vertex. It is also called the weight of an edge.

**Comment by KrishnaAlagiri** [comment]
All issues resolved.
PS: Pardon the delay, laptop went for conditioning and I'm working on a rpi0 :33

**Pull request merged**